server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -6408,7 +6408,9 @@ public NetworkOfferingVO doInTransaction(final TransactionStatus status) {
                             detailsVO.add(new NetworkOfferingDetailsVO(offering.getId(), Detail.internetProtocol, String.valueOf(internetProtocol), true));
                         }
                         if (!detailsVO.isEmpty()) {
-                            networkOfferingDetailsDao.saveDetails(detailsVO);
```

Comment:
`saveDetails` will firstly remove all existing details. this causes issue (2) in the PR decription.

in my opinion. the issue is critical . @DaanHoogland @rohityadavcloud @shwstppr

Reply:
@weizhouapache this is used while creating a new offering so will it matter even if the code tries to delete existing details for the resourceId (service offering is here)?

Reply:
> @weizhouapache this is used while creating a new offering so will it matter even if the code tries to delete existing details for the resourceId (service offering is here)?

@shwstppr yes, there are some network offering details passed via API.
https://cloudstack.apache.org/api/apidocs-4.17/apis/createNetworkOffering.html

![image](https://user-images.githubusercontent.com/57355700/210777355-87067e0a-b0ef-412f-9514-345ff07b4260.png)


for example, the following details are not saved

Promiscuous mode
MAC address changes
Forged transmits
MAC learning